### PR TITLE
#CP-28 feat: CRUD operations for buses integration

### PR DESCRIPTION
**Pull request by Yvad60**
#### What does this PR do?
allows the operator/admin to be able to register edit and delete buses using real data from the backend

#### Description of the tasks to be completed
- [x] The operator should be able to create, update, and delete a bus.
- [x] The operator should be able to view a paginated list of buses in the system.
- [x] Users should be able to view the bus's details.
- [x] All Requests should be sent to the backend and return appropriate responses
- [x] Appropriate error messages are displayed on unsuccessful attempts
- [x] The system should work with real data from the database

#### How should this be manually tested?

```
git fetch
git checkout ft-crud-operations-for-buses-cp-28
npm install
npm start 
login as an admin to be able to access buses
test bus cruds functionality
```

#### Pre-requisites:

- Clone the project

#### What are the relevant trello story:
[#CP-28](https://trello.com/c/0JmHZliU/28-crud-operations-for-buses)

### CODE REVIEW

**Review by samyshyaka — COMMENTED** [review]
Nice work @Yvad60, The code base looks good and the feature is working as expected. After you finish working on the tests, I believe the PR will be ready to be merged.

**Review by feyton — DISMISSED** [review]
### Approval
@Yvad60 this is great work, I am approving your work as the feature is well implemented and works as expected.

File: src/components/dropdowns/BusManagement.js [hunk 1]
```diff
@@ -13,25 +14,38 @@ const BusManagement = ({ row }) => {
...
-    try {
-      await axios.delete(`/buses/${busId}`);
-      toast('Bus deleted successfully', { type: 'success' });
-      setloading(false);
-      selectRef.current.innerHTML = `<b className="deleted mx-auto font-bold">Deleted</>`;
-    } catch (error) {
-      toast('Error occured', { type: 'error' });
-      setloading(false);
+  const handleDelete = async (busInfo) => {
+    const results = await Swal.fire({
```
Comment (Jkadhuwa) [inline src/components/dropdowns/BusManagement.js:20]: You can create a custom hook to handle all the delete method and call it inside the handleDelete(); see implementation below.

https://reactjs.org/docs/hooks-custom.html

File: src/components/forms/Buses.js [hunk 2]
```diff
@@ -22,9 +25,29 @@ const Buses = ({
...
-    await formAction(data);
-    setloading(false);
+    try {
+      setloading(true);
+      await formAction(data);
+      setloading(false);
+      navigate(-1);
+    } catch (error) {
+      const validationErrors = error?.response?.data.data;
+      if (validationErrors) {
```
Comment (Jkadhuwa) [inline src/components/forms/Buses.js:37]: Avoid nesting  of conditional operators and also you can move this to a custom hook to increase readability and reduce complexity

**Review by Jkadhuwa — CHANGES_REQUESTED** [review]
Good work, solve codeclimate issues and increase coverage and it will be good for merging.
Refer here to test Axios requests.
https://betterprogramming.pub/testing-react-list-using-axios-and-react-testing-library-d000eebf3413

**Comment by feyton** [comment]
This PR has been implemented on another branch

**Pull request closed without merging**